packages/nx-semantic-release/src/common/git.ts

```diff
@@ -82,12 +83,28 @@ export function shouldSkipCommit(
 async function listAffectedFilesInCommit(
   commit: Pick<Commit, 'commit'>
 ): Promise<string[]> {
+  // eg. /code/Repo/frontend/
+  const cwd = process.cwd() + '/';
+  // eg. /code/Repo/
+  const repositoryRoot = await exec('git rev-parse --show-toplevel') + '/';
+  // Matches the start of a path from the git root to the nx root
+  const nxPathPart = new RegExp(`^${cwd.substring(repositoryRoot.length)}`);
+
   const files = await exec(`git show --name-status ${commit.commit.short}`);
 
   return files
     .toString()
     .split('\n')
     .filter(Boolean)
     .map((line) => line.split('\t')[1])
-    .filter(Boolean);
+    .filter(Boolean)
+    .filter((filePath: string) => {
+      // only include files inside the nx root
+      return filePath.match(nxPathPart);
+    })
+    .map((filePath: string) => {
+        // The filepaths start from the root of the git repository, but
+        // in our case we want them to start from the nx root.
+        return filePath.replace(nxPathPart, '');
+    });
```

Comment:
I would suggest one small change:

```typescript
return files
    .toString()
    .split('\n')
    .map((line) => line?.split('\t')?.[1])
    .filter(Boolean)
    .filter((filePath: string) => {
      // only include files inside the nx root
      return filePath.match(nxPathPart);
    })
    .map((filePath: string) => {
        // The filepaths start from the root of the git repository, but
        // in our case we want them to start from the nx root.
        return filePath.replace(nxPathPart, '');
    });
```

With that we can avoid calling `.filter(Boolean)` twice 😄

Reply:
Ah yeah good one :)

Updated and tested, still working 👍